src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -3991,66 +3995,70 @@ class CgroupUtils(object):
             self._delete_slice(jobid)
             return True
         # Cgroup removal has failed
-        pbs.logmsg(pbs.EVENT_SYSTEM, "cgroup still has %d tasks: %s" %
+        pbs.logmsg(pbs.EVENT_SYSTEM, 'cgroup still has %d tasks: %s' %
                    (remaining, parent))
         if not do_offline:
-            pbs.logmsg(pbs.EVENT_DEBUG4, "%s: Offline not requested" %
+            pbs.logmsg(pbs.EVENT_DEBUG4, '%s: Offline not requested' %
                        caller_name())
             return False
         # Rerun the job and log the message
-        pbs.logmsg(pbs.EVENT_DEBUG, '%s: Failed to cleanup cgroup for %s' %
-                   (caller_name(), jobid))
+        if jobid:
+            pbs.logmsg(pbs.EVENT_DEBUG, '%s: Failed to cleanup cgroup for %s' %
+                       (caller_name(), jobid))
+            pbs.logmsg(pbs.EVENT_DEBUG2, '%s: Job %s will be requeued' %
+                       (caller_name(), jobid))
+        else:
+            pbs.logmsg(pbs.EVENT_DEBUG, '%s: Failed to cleanup %s' %
+                       (caller_name(), path))
         pbs.logmsg(pbs.EVENT_DEBUG2, '%s: Taking node offline' %
                    caller_name())
-        pbs.logmsg(pbs.EVENT_DEBUG2, '%s: Job %s will be requeued' %
-                   (caller_name(), jobid))
         # Check to see if the offline file is already present
         if os.path.isfile(self.offline_file):
-            msg = "Cgroup(s) not cleaning up but the node already "
-            msg += "has the offline file."
-            pbs.logmsg(pbs.EVENT_DEBUG2, msg)
+            pbs.logmsg(pbs.EVENT_DEBUG2,
+                       'Cgroup(s) not cleaning up but the node already ' +
```

Comment:
Remove this +.  It isn't needed and will cause the string to be regenerated.

Reply:
Fixed several instances, including those you flagged.